File: 844-backspace-string-compare/JavaScript/1-backspace-string-compare[brute-force].js
const s = 'ab#c';
t = 'ad#c';

const buildString = (string) => {
  const buildString = [];
  for (let p = 0; p < string.length; p++) {
    string[p] !== '#' ? buildString.push(string[p]) : buildString.pop();
  }
  return buildString;
};

var backspaceCompare = function (s, t) {
  const finalS = buildString(s);
  const finalT = buildString(t);

  if (finalS.length !== finalT.length) {
    return false;
  } else {
    for (let p = 0; p < finalS.length; p++) {
      if (finalS[p] !== finalT[p]) {
        return false;
      }
    }
  }
  return true;
};

console.log(backspaceCompare(s, t));

File: 844-backspace-string-compare/Python/1-backspace-string-compare[brute-force].py
s = 'ab#c'
t = 'ad#c'

class Solution:
    def build_string(self,string):
      built_string = []
      for s in string:
        if s != '#':
          built_string.append(s)
        elif built_string:
          built_string.pop()
      return built_string

    def backspaceCompare(self, s: str, t: str) -> bool:
      final_s = self.build_string(s)
      final_t = self.build_string(t)

      if len(final_s) != len(final_t):
        return False
      else:
        for fs in range(len(final_s)):
          if final_s[fs] != final_t[fs]:
            return False
      return True



solution = Solution()

print(solution.backspaceCompare(s, t))